### PR TITLE
[MetricBeat] support wildcard `*`  dimension value in AWS CloudWatch module

### LINKED ISSUE #18050: [MetricBeat] AWS module support wildcard format dimension value to reduce the metrics count
##  1. Describe the enhancement:

AWS CloudWatch metrics have many dimensions, for example,  In `AWS/NetworkELB`  namespace, `UnHealthyHostCount` has the dimensions:

* TargetGroup, LoadBalancer
*  TargetGroup, LoadBalancer, AvailableZone

When we query CloudWatch metrics with GetMetricData API,  there are two metric.

```
        {
            "Namespace": "AWS/NetworkELB", 
            "Dimensions": [
                {
                    "Name": "TargetGroup", 
                    "Value": "targetgroup/some-name/222222"
                }, 
                {
                    "Name": "AvailabilityZone", 
                    "Value": "ap-northeast-1a"
                }, 
                {
                    "Name": "LoadBalancer", 
                    "Value": "net/some-lb-name/1111111"
                }
            ], 
            "MetricName": "UnHealthyHostCount"
        },

                {
            "Namespace": "AWS/NetworkELB", 
            "Dimensions": [
                {
                    "Name": "TargetGroup", 
                    "Value": "targetgroup/some-name/222222"
                }, 
                {
                    "Name": "LoadBalancer", 
                    "Value": "net/some-lb-name/1111111"
                }
            ], 
            "MetricName": "UnHealthyHostCount"
        }, 
```

---

If metricbeat could limit dimension names scope, after compared to `ListMetrics API` result, just one metric is retrived from AWS. so this can reduce `AWS GetMetircData API ` cost and metricbeat events amount.

```yaml
- module: aws
  period: 300s
  metricsets:
    - cloudwatch
  metrics:
    - namespace: AWS/NetworkELB
      statistic: ["Maximum"]
      name: ["UnHealthyHostCount"]
     # demo param
      dimension_names: ["LoadBalancer", "TargetGroup"]
      tags.resource_type_filter: elasticloadbalancing

```

metricbeat will only make one metric query: 

```
        {
            "Namespace": "AWS/NetworkELB", 
            "Dimensions": [
                {
                    "Name": "TargetGroup", 
                    "Value": "targetgroup/some-name/222222"
                }, 
                {
                    "Name": "LoadBalancer", 
                    "Value": "net/some-lb-name/1111111"
                }
            ], 
            "MetricName": "UnHealthyHostCount"
        }, 
```

## 2. Describe a specific use case for the enhancement or feature:

* reduce AWS GetMetricData cost
* reduce metricbeat events amount 

## 3. reference

* aws doc:

https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-cloudwatch-metrics.html

* inspired by:

https://github.com/prometheus/cloudwatch_exporter#cost

### PR DESCRIPTION
**Pull request by kwinstonix**
- Enhancement

## What does this PR do?

* support wildcard `*` dimension value in AWS CloudWatch metrics config
* update document with example config

## Why is it important?

Now we must set explicit dimension value to filter out CloudWatch metrics. It will be flexible if we can support wildcard `*` in value param.

current config example:
```yaml
- module: aws
  period: 300s
  metricsets:
    - cloudwatch
  metrics:
    - namespace: AWS/ApplicationELB
      statistic: ['Maximum']
      name: ['UnHealthyHostCount']
      dimensions:
        - name: LoadBalancer
          value: "lb id"
        - name: TargetGroup
          value: "tg id"
```

with the enhancement:

```yaml
- module: aws
  period: 300s
  metricsets:
    - cloudwatch
  metrics:
    - namespace: AWS/ApplicationELB
      statistic: ['Maximum']
      name: ['UnHealthyHostCount']
      dimensions:
        - name: LoadBalancer
          value: "*"
        - name: TargetGroup
          value: "*"

```

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes elastic/beats#18050

### CODE REVIEW

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -589,22 +604,37 @@ func reportEvents(eventsWithIdentifier map[string]mb.Event, report mb.ReporterV2
 	return nil
 }
 
+func configDimensionValueContainsWildcard(dim []Dimension) bool {
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go:607]: What you think about adding a unit test for this function?

**Comment by kaiyan-sheng** [comment]
jenkins run tests please

**Comment by kwinstonix** [comment]
Ok, I will add that unit test, thank you

**Comment by kaiyan-sheng** [comment]
jenkins run the tests

**Comment by kwinstonix** [comment]
jenkins run the tests

**Comment by kaiyan-sheng** [comment]
@kwinstonix Thank you! I just resolved the merge conflict and once the CI gives a green, I will merge this PR and backport it. Thanks again for your contribution!!

**Pull request merged**